### ISSUE TITLE
impossible to launch several containers of the same project at once

### ISSUE THREAD

**Issue by rvalyi** [akretion/voodoo-template#5]
there is the port clash which would be easy to fix by offsetting things.

there is a name clash as the container is named after the folder name. We should eventually test if there is such docker container and if so increment some counter.

finally is that going to work to have 2 instance of a Postgresql server using the same data dir? I would say yes, but that's worth checking.

**Comment by fgrehm** [comment]
> finally is that going to work to have 2 instance of a Postgresql server using the same data dir? I would say yes, but that's worth checking.

I remember trying that out and it didn't work, but please tell me if you are able to do that

**Comment by rvalyi** [comment]
Hey, hello Fabio.

That's amazing I just tried over the last hour and came to the same conclusion: it's talking about CRITICAL flush errors in the Postgresql log.
Too bad, I'll have to prevent creating 2 containers of the same project again. Unless I try to inject the Postgresql of the 1st container into the 2nd with --link but as for me this is too much hassle for too little benefit.

BTW, I just started basing my image on your official devstep image now (I as using a fork before you released 0.1.0)

**Comment by fgrehm** [comment]
One option would be to manage a separate postgres container and always use `--link`s + Devstep's default port forwarding, but I understand you want to keep things simpler and avoid using multiple containers.

**Comment by rvalyi** [comment]
Yes, in my use case the benefit of simplicity largely outweighs the impossibility to have 2 containers.

I don't know if that's a use case you see for Devstep. But as I said, I see a use case embedding it in Vagrant for the disabled part of the humanity. Unless I see strong signals Vagrant start managing the whole Docker cycles flawlessly, I prefer a single container.

Also, I personally see room to use Postgresql just like Sqlite here. But again, this may be the specificity of the project I deal with that won't work with Sqlite while standard projects tend to use decent ORM's that work with Sqlite for development...

**Comment by fgrehm** [comment]
Just so you know, the support for running commands on an existing container might land in docker soon, there is a PR open for that and it is likely that it'll make things easier for you ;-)
